libraries/entities/src/index.js

```diff
@@ -23,7 +26,54 @@ class Entity {
 
   create(properties, overwrite) {
     const documentId = this.calculateId(properties);
-    this.db.procs[this.serviceName][`${this.tableName}_create`](documentId, properties, overwrite, 1);
+
+    return this.db.procs[`${this.tableName}_create`](documentId, properties, overwrite, 1);
+  }
+
+  delete(properties) {
+    const documentId = this.calculateId(properties);
+
+    return this.db.procs[`${this.tableName}_delete`](documentId);
+  }
+
+  update(properties) {
+    const documentId = this.calculateId(properties);
+
+    return this.db.procs[`${this.tableName}_update`](documentId, properties, 1);
+  }
+
+  load(properties) {
+    const documentId = this.calculateId(properties);
+
+    return this.db.procs[`${this.tableName}_load`](documentId);
+  }
+
+  async scan(options = {}) {
+    const {
+      filter,
+      limit = 1000,
+      page = 1,
+    } = options;
+
+    if (filter) {
+      assert(typeof filter === 'function', 'filter should be a function');
+    }
+
+    const result = await this.db._withClient(READ, client => {
+      return client.query(`select * from ${this.tableName}`);
+    });
+
+    const filteredRows = filter ? result.rows.filter(filter) : result.rows;
```

Comment:
We probably can't get away with doing client-side filtering -- that will have the database sending the whole table to us every time.  We'll need to find a way to translate at least the scan conditions related to the row and partition keys into a `WHERE` clause.